packages/rum-core/src/domain/view/viewMetrics/trackScrollMetrics.spec.ts

```diff
@@ -26,117 +39,85 @@ describe('trackScrollMetrics', () => {
         viewTest = setupViewTest(buildContext)
         return viewTest
       })
+    callbackSpy = jasmine.createSpy('callback')
+    stopTrackScrollMetrics = trackScrollMetrics(
+      {} as RumConfiguration,
+      { relative: 0 as RelativeTime, timeStamp: 0 as TimeStamp },
+      callbackSpy,
+      getMetrics,
+      0
+    ).stop
+    document.body.style.margin = '0'
+
+    assertCallbackSpyEqual = (expected, done, times = 1) => {
+      collectAsyncCalls(callbackSpy, times, (calls) => {
+        expect(calls.mostRecent().args[0]).toEqual(expected)
+        done()
+      })
+    }
   })
 
   afterEach(() => {
+    stopTrackScrollMetrics()
+    document.body.replaceChildren()
     setupBuilder.cleanup()
   })
 
-  describe('on scroll', () => {
-    let scrollMetrics: ScrollMetrics | undefined
-    let stopTrackScrollMetrics: () => void
-    let clock: Clock
-    let configuration: RumConfiguration
-
-    const getMetrics = jasmine.createSpy('getMetrics')
-
-    const newScroll = (scrollParams: { scrollHeight: number; scrollDepth: number; scrollTop: number }) => {
-      getMetrics.and.returnValue(scrollParams)
-
-      window.dispatchEvent(createNewEvent(DOM_EVENT.SCROLL))
-
-      clock.tick(THROTTLE_SCROLL_DURATION)
-    }
-
-    beforeEach(() => {
-      configuration = {} as RumConfiguration
-      clock = mockClock()
-      stopTrackScrollMetrics = trackScrollMetrics(
-        configuration,
-        { relative: 0 as RelativeTime, timeStamp: 0 as TimeStamp },
-        (s) => (scrollMetrics = s),
-        getMetrics
-      ).stop
-    })
-
-    afterEach(() => {
-      stopTrackScrollMetrics()
-      scrollMetrics = undefined
-      clock.cleanup()
-    })
-
-    it('should update scroll metrics when scrolling the first time', () => {
-      newScroll({ scrollHeight: 1000, scrollDepth: 500, scrollTop: 100 })
+  it('should update scroll height if resize occurred before scroll', (done) => {
+    increaseHeight({ scrollDepth: 0, scrollHeight: 2000, scrollTop: 0 })
+    assertCallbackSpyEqual(
+      {
+        maxDepthScrollHeight: 2000,
+        maxDepth: 0,
+        maxDepthScrollTop: 0,
+        maxDepthTime: jasmine.any(Number),
+      },
+      done
+    )
+  })
 
-      expect(scrollMetrics).toEqual({
-        maxDepthScrollHeight: 1000,
+  it('should update scroll depth when scrolling the first time', (done) => {
+    newScroll({ scrollHeight: 1000, scrollDepth: 500, scrollTop: 100 })
+    assertCallbackSpyEqual(
+      {
         maxDepth: 500,
         maxDepthScrollTop: 100,
-        maxDepthTime: 1000 as Duration,
-      })
-    })
-
-    it('should update scroll metrics when scroll depth has increased', () => {
-      newScroll({ scrollHeight: 1000, scrollDepth: 500, scrollTop: 100 })
+        maxDepthScrollHeight: 0,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** `maxDepthScrollHeight` being `0` is a bit unexpected: `maxDepthScrollHeight` should always be greater than or equal to `maxDepth` right?